src/NFSFile.cpp

```diff
@@ -596,7 +597,7 @@ bool CNFSFile::GetDirectory(const VFSURL& url, std::vector<kodi::vfs::CDirEntry>
       P8PLATFORM::FileTimeToLocalFileTime(&fileTime, &localTime);*/
 
       kodi::vfs::CDirEntry pItem;
-      pItem.SetLabel(strdup(nfsdirent->name));
+      pItem.SetLabel(strdup(tmpDirent.name));
```

Comment:
Leaks

Reply:
how?

Reply:
strdup() calls malloc, passes the memory to the std::string constructor, then fails to free the memory. This line should be

```c++
pItem.SetLabel(tmpDirent.name);
```

or 

```c++
pItem.SetLabel(tmpDirent.name != nullptr ? tmpDirent.name : "");
```

Reply:
ah ok, I image I should do the same with the other strdup